base/pointer.jl

```diff
@@ -98,32 +98,146 @@ unsafe_wrap(Atype::Union{Type{Array},Type{Array{T}},Type{Array{T,N}}},
 
 """
     unsafe_load(p::Ptr{T}, i::Integer=1)
+    unsafe_load(p::Ptr{T}, order::Symbol)
+    unsafe_load(p::Ptr{T}, i::Integer, order::Symbol)
 
 Load a value of type `T` from the address of the `i`th element (1-indexed) starting at `p`.
-This is equivalent to the C expression `p[i-1]`.
+This is equivalent to the C expression `p[i-1]`. Optionally, an ordering can be provided.
```

Comment:
Instead of `"an ordering"` can you say that it is the "the atomic memory ordering". Or, more verbosely, to borrow from C++:
> specifies how memory accesses, including regular, non-atomic memory accesses, are to be ordered around an atomic operation

To say that "Optionally, the `order` argument can be used to specify whether other memory access are permitted to be reordered around this operation when they are visible to other threads."